packages/build/src/core/constants.js

```diff
@@ -13,7 +13,7 @@ const getConstants = async function ({
   buildDir,
   functionsDistDir,
   netlifyConfig: {
-    build: { publish = buildDir, functions, edge_handlers: edgeHandlers },
+    build: { publish, functions, edge_handlers: edgeHandlers },
```

Comment:
Not needed anymore since `build.publish` is never `undefined` anymore (it always defaults to `buildDir`).